test/integration/dedupe.spec.js

```diff
@@ -198,5 +200,22 @@ describe('Request deduping', function() {
             ]);
         }, done, partDone);
     });
+
+    it("deduplicates gets with overlapping ranges", function (done) {
+        var onGet = sinon.spy();
+        var model = new falcor.Model({ source: new LocalDataSource(cacheGenerator(0, 3), { wait: 0, onGet: onGet }) });
+
+        var partDone = after(3, function() {
+            expect(onGet.getCall(0).args[1]).to.deep.equal([["videos", 0, "title"]]);
```

Comment:
Do we really care what order the `onGet` calls were in? Wouldn't we be happy with these calls occurring in any order? If so, can we make a looser assertion that's less brittle?

Reply:
I'll see if I can do that without getting too complicated.

For reference, some functionality in Falcor is explicitly synchronous, such as returning cached data immediately. I don't believe that includes outgoing requests, however. (As far as Falcor knows, it's an implementation detail of DataSource implementations.)

Reply:
well since you're not enabling batching in the `model` here, it is expected that the requests will be made in sequence. i guess you could say that we don't explicitly say this in the contract between falcor and datasource, so it's just an _expectation_

Reply:
Decided not to mess with it right now. Call order is due to LocalDataSource implementation, which was specifically written for the test suite.